programs/account-compression/src/instructions/nullify_leaves.rs

```diff
@@ -28,33 +28,63 @@ pub struct NullifyLeaves<'info> {
 pub fn process_nullify_leaves<'a, 'b, 'c: 'info, 'info>(
     ctx: &'a Context<'a, 'b, 'c, 'info, NullifyLeaves<'info>>,
     change_log_indices: &'a [u64],
-    leaves_indices: &'a [u16],
+    leaves_queue_indices: &'a [u16],
     indices: &'a [u64],
     proofs: &'a [Vec<[u8; 32]>],
 ) -> Result<()> {
-    // TODO: activate when group access control is implemented
-    // check_registered_or_signer::<AppendLeaves, StateMerkleTreeAccount>(
-    //     &ctx,
-    //     &merkle_tree_account,
-    // )?;
     let mut array_account = ctx.accounts.indexed_array.load_mut()?;
-    let leaf: [u8; 32] = array_account.indexed_array[leaves_indices[0] as usize].element;
+    let mut merkle_tree_account = ctx.accounts.merkle_tree.load_mut()?;
+    if array_account.associated_merkle_tree != ctx.accounts.merkle_tree.key() {
+        msg!(
+            "merkle tree and indexed array are not associated {} != {}",
+            array_account.associated_merkle_tree,
+            ctx.accounts.merkle_tree.key()
+        );
+        return Err(AccountCompressionErrorCode::InvalidMerkleTree.into());
+    }
+    if merkle_tree_account.associated_queue != ctx.accounts.indexed_array.key() {
+        msg!(
+            "merkle tree and indexed array are not associated {} != {}",
```

Comment:
"provided merkle tree [...]" ?; also may make sense to consolidate the names for queue and indexed_array / or add a note saying that it's the same

Reply:
Changed it in the logs, I want to do a separate pr to change the variables.